src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+                + "JOIN cars_drivers AS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE driver_id = ? AND is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't get all cars by driver with id:" + driverId + ". ", e);
+        }
+        cars.stream().forEach(c -> c.setDrivers(getDriversForCar(c.getId())));
+        return cars;
+    }
+
+    private void updateDriverList(Car car) {
+        String deleteQuery = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(deleteQuery)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update (delete)) "
+                    + car + " in cars DB.", e);
+        }
+        if (!car.getDrivers().isEmpty()) {
+            String insertQueryValues = car.getDrivers()
+                    .stream()
+                    .map(d -> "('" + d.getId() + "', '" + car.getId() + "')")
+                    .collect(Collectors.joining(", "));
+            String insertQuery =
+                    "INSERT INTO cars_drivers (`driver_id`, `car_id`) VALUES "
+                            + insertQueryValues + ";";
+            try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement
+                            = connection.prepareStatement(insertQuery)) {
+                statement.executeUpdate();
+            } catch (SQLException e) {
+                throw new DataProcessingException("Couldn't update (add)"
+                        + car + " in cars DB.", e);
+            }
+        }
+    }
+
+    private void insertDrivers(Car car) {
+        String request = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(request)) {
+            for (Driver c : car.getDrivers()) {
+                statement.setLong(1, c.getId());
+                statement.setLong(2, car.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException err) {
+            throw new RuntimeException("Can't add car drivers to `cars_drivers`", err);
+        }
+    }
+
```

Comment:
would suggest creating DriverDao for manipulation driver' db.

Reply:
and there execute queries related to driver' table.

Reply:
DriverDao is for drivers table only, no joins. In the task we nned to operate with table cars_drivers